### ISSUE TITLE
Can't install on Linux due to permission denied on ~/.node-gyp

### ISSUE THREAD

**Issue by dandv** [pnpm/pnpm#1030]
### pnpm version:

I can't install pnpm on Ubuntu 16.04 with node 9.4.0 as sudo.

### Code to reproduce the issue:

```
$ sudo npm install -g pnpm
/usr/bin/pnpm -> /usr/lib/node_modules/pnpm/lib/bin/pnpm.js
/usr/bin/pnpx -> /usr/lib/node_modules/pnpm/lib/bin/pnpx.js

> drivelist@5.2.12 install /usr/lib/node_modules/pnpm/node_modules/drivelist
> prebuild-install || node-gyp rebuild

prebuild-install info begin Prebuild-install version 2.5.0
prebuild-install info looking for local prebuild @ prebuilds/drivelist-v5.2.12-node-v59-linux-x64.tar.gz
prebuild-install WARN install EACCES: permission denied, access '/home/dandv/.npm'
gyp WARN EACCES user "root" does not have permission to access the dev dir "/home/dandv/.node-gyp/9.5.0"
gyp WARN EACCES attempting to reinstall using temporary dev dir "/usr/lib/node_modules/pnpm/node_modules/drivelist/.node-gyp"
gyp WARN EACCES user "root" does not have permission to access the dev dir "/usr/lib/node_modules/pnpm/node_modules/drivelist/.node-gyp/9.5.0"
```

The above error keeps getting output until I terminate the script. Thus I can't upload a gist of the pnpm-debug.log file.

The solution was to delete `~/.node-gyp`, but I don't know why as sudo, the installer wasn't able to do that.

**Comment by zkochan** [comment]
A workaround might be to run `sudo npm install -g pnpm --ignore-scripts`

The only issue is that in this case, pnpm won't be able to detect the disks on the machine. So it won't be able to automatically select a store location that is on the same disk. Of course, you can always manually specify the store location.

**Comment by zkochan** [comment]
Isn't this a duplicate of #1034? anyway, this should be fixed from pnpm v1.33.0 as it has not native deps anymore
